pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -98,13 +99,46 @@ func fixKvmEnabled() error {
 	return nil
 }
 
+func getLibvirtCapabilities() (*libvirtxml.Caps, error) {
+	stdOut, _, err := crcos.RunWithDefaultLocale("virsh", "capabilities")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to run 'virsh capabilities': %v", err)
+	}
+	caps := &libvirtxml.Caps{}
+	err = caps.Unmarshal(stdOut)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing 'virsh capabilities': %v", err)
+	}
+
+	return caps, nil
+}
+
 func checkLibvirtInstalled() error {
 	logging.Debug("Checking if 'virsh' is available")
 	path, err := exec.LookPath("virsh")
 	if err != nil {
 		return fmt.Errorf("Libvirt cli virsh was not found in path")
 	}
 	logging.Debug("'virsh' was found in ", path)
+
+	logging.Debug("Checking 'virsh capabilities' for libvirtd/qemu availability")
+	caps, err := getLibvirtCapabilities()
+	if err != nil {
+		return err
+	}
+
+	foundHvm := false
+	for _, guest := range caps.Guests {
+		if guest.OSType == "hvm" && guest.Arch.Name == "x86_64" {
```

Comment:
This check will fail on a different Arch like ppc64 if someone build the crc on that platform, instead of hard coding `x86_64` shouldn't we use `runtime.GOARCH` ?

Reply:
Yup, I was a bit worried about this. `runtime.GOARCH` returns `amd64` for `x86_64`, so we need to do some mapping from goarch to libvirt arch. When adding a new arch, it will be failing until we add the new mapping. I think what I can do is read `host.cpu.arch` from `virsh capabilities` and match it here.

Reply:
you mean `guest.Arch.Name === host.CPU.Arch` which means what ever the host arch should be guest one, that seems bit clean.

Reply:
Done in the new revision